### PR TITLE
Jinja template prototype 

### PR DESCRIPTION
**Pull request by ocefpaf**
@rsignell-usgs this is not ready yet. It is a prototype to use jinja  templates to generate the `ncml`.

Please take a look and let me know if you think it is worth pursuing.

### CODE REVIEW

File: yaml2ncml/yaml2ncml.py [hunk 1]
```diff
@@ -29,61 +32,16 @@
...
-            email = value.get("email", None)
-            if email:
-                text += str_att('_'.join([key, 'email']), email)
-            url = value.get("url", None)
-            if url:
-                text += str_att('_'.join([key, 'url']), url)
-            name = value.get("name", None)
-            if name:
-                text += str_att('_'.join([key, 'name']), name)
-    return text
```
Comment (ocefpaf) [inline yaml2ncml/yaml2ncml.py]: `header` and `add_global_atts` are parsed with the template. The renaming fields will be added later.

File: yaml2ncml/yaml2ncml.py [hunk 2]
```diff
@@ -224,14 +182,42 @@ def add_aggregation_scan(text, a):
...
+        ('_'.join([name, key]), attrs[name].get(key))
+        for key in attrs[name].keys() if attrs[name].get(key)
+    )
+
+
+templ = Template("""<?xml version="1.0" encoding="UTF-8"?>
+<netcdf xmlns="http://www.unidata.ucar.edu/namespaces/netcdf/ncml-2.2">
+{% for name, value in global_attrs | dictsort %}  <attribute name="{{ name }}" type="String" value="{% if value is string %}{{ value }}{% else %}{{ value|join(',') }}{%endif%}"/>
+{% endfor %}
+""")  # noqa
```
Comment (ocefpaf) [inline yaml2ncml/yaml2ncml.py]: This is the template. It is basically a for-loop that adds the `name`, `value` for the many global attributes.

**Comment by rsignell-usgs** [comment]
Filipe, I think I'll have to get you to explain this to me tomorrow.   I don't really get it.

**Comment by rsignell-usgs** [comment]
Since this PR I now understand templates,  this approach makes all kinds of sense.   It would make the code way more easy to maintain and improve. 

@ocefpaf , does this still need work, or could @zdefne-usgs give it a try?

**Comment by ocefpaf** [comment]
> @ocefpaf , does this still need work

This PR is more than 1 year old... So probably.

> or could @zdefne-usgs give it a try?

That would be awesome. I'll try to get back to this PR by the end of the week.

**Comment by rsignell-usgs** [comment]
On further reflection, it's probably best to let @zdefne-usgs focus on getting his NcML and ISO records done for the data publication he is working on.   He's trying to get the THREDDS-generated ISO records to the point where they are accepted by our USGS metadata reviewers

**Comment by zdefne-usgs** [comment]
I'll be happy to work on this as soon as I submit my files for a review. I will probably need to discuss it with @rsignell-usgs next week to catch up with what need to be done.

**Pull request closed without merging**